pkg/pool-linear/contracts/LinearPool.sol

```diff
@@ -395,7 +395,7 @@ abstract contract LinearPool is ILinearPool, IGeneralPool, IRateProvider, BasePo
         address recipient,
         uint256[] memory,
         bytes memory
-    ) internal view override whenNotPaused returns (uint256, uint256[] memory) {
```

Comment:
Linear Pools also pause on swaps

Reply:
Yes; is something wrong here? onSwap still has the whenNotPaused modifier, because LinearPool doesn't inherit from BaseGeneralPool (only implements the interface). But it does in inherit from BasePool, and BasePool.onJoinPool blocks all joins on pause (including initialization), so we don't need it here